File: CreationalPatterns/AbstractFactory/C#/Vehicle Factory/ConcreteFactories/JapaneseVehicleFactory.cs
using Vehicles.Interfaces;
using Vehicles.Products;

namespace Vehicles.ConcreteFactories
{
    public class JapaneseVehicleFactory : IVehicleFactory
    {
        public IVehicle CreateSedan()
        {
            return new JapaneseSedan();
        }

        public IVehicle CreateSUV()
        {
            return new JapaneseSUV();
        }

        public IVehicle CreateSportsCar()
        {
            return new JapaneseSportsCar();
        }
    }
}

File: CreationalPatterns/AbstractFactory/Java/Vehicle/src/concretefactories/JapaneseVehicleFactory.java
package concretefactories;

import interfaces.IVehicle;
import interfaces.IVehicleFactory;
import products.JapaneseSUV;
import products.JapaneseSedan;
import products.JapaneseSportsCar;

public class JapaneseVehicleFactory implements IVehicleFactory {

    @Override
    public IVehicle createSedan() {
        return new JapaneseSedan();
    }

    @Override
    public IVehicle createSUV() {
        return new JapaneseSUV();
    }

    @Override
    public IVehicle createSportsCar() {
        return new JapaneseSportsCar();
    }
}